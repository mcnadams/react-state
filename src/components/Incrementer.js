import React, { PureComponent } from 'react';

export default class Incrementer extends PureComponent {
  state = {
    count: 0
  };

  incrementCounter = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  }

  render() {
    return (
      <>
        <div>Current Count: { this.state.count }</div>
        <button onClick={this.incrementCounter}>Increment</button>
      </>
    );
  }

}
