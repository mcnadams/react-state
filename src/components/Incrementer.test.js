import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('Simple Incrementer', () => {
  it('makes an incrementer button', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });

});
