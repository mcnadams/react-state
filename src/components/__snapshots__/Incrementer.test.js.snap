// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Incrementer makes an incrementer button 1`] = `
<Fragment>
  <div>
    Current Count: 
    0
  </div>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</Fragment>
`;
